processor/src/main/java/org/mapstruct/ap/internal/model/source/builtin/BuiltInMethod.java

```diff
@@ -141,6 +141,11 @@ public Parameter getTargetTypeParameter() {
         return null;
     }
 
+    @Override
+	public boolean isObjectFactory() {
+		return getTargetTypeParameter() != null || getParameters().isEmpty();
```

Comment:
just return `false` here. We have no built-in methods that act as factory methods.